conf/cassandra.yaml

```diff
@@ -2076,6 +2076,15 @@ drop_compact_storage_enabled: false
 # if zero_ttl_on_twcs_enabled is set to false, this property is irrelevant as such statements will fail.
 #zero_ttl_on_twcs_warned: true
 
+# Guardrail enabling querying a table with indexes without specifying partition keys, defaults to true
+#non_partition_key_restricted_query_enabled: true
+# Maximum number of referenced indexes on a replica when executing a SELECT query without partition key restrictions
```

Comment:
nit: In our inline comments, we should probably make it clear that this is an SAI-specific thing. SASI could have used it too, in theory, but SAI should be completely replacing it, so nobody cares.

Reply:
@maedhroz , well ... we plan to cover old / original secondary indexes too, no? Then the message not referencing SAI just probably just fine?

Reply:
No, the concept of SSTable attachment and therefore SSTable indexes is specific to SAI (and SASI, but we don't care). The partition restriction guardrail is the one that applies to all 2i implementations.